website/addons/s3/static/s3NodeConfig.js

```diff
@@ -91,31 +91,45 @@ ViewModel.prototype.updateBucketList = function(){
 
 ViewModel.prototype.selectBucket = function() {
     var self = this;
+
+    // The current version of boto does not handle bucket names containing periods
+    var isValidBucket = /^(?!.*(\.\.|-\.))[^.][a-z0-9\d-]{2,61}[^.]$/;
+
     self.loading(true);
-    return $osf.postJSON(
+
+    if (isValidBucket.exec(self.selectedBucket()) == null) {
```

Comment:
Why not make `isValidBucket` a function, so that we can more easily change it's implementation in the future?